pkg/services/rendering/rendering.go

```diff
@@ -86,12 +86,23 @@ func ProvideService(cfg *setting.Cfg, remoteCache *remotecache.RemoteCache, rm p
 		domain = "localhost"
 	}
 
-	s := &RenderingService{
-		perRequestRenderKeyProvider: &perRequestRenderKeyProvider{
+	var renderKeyProvider renderKeyProvider
```

Comment:
Also thought about adding this handling inside the `perRequestRenderKeyProvider` implementation, but finally decided that follow this approach cause it looks cleaner (more isolated). Anyway, any kind of feedback will be more than welcome.